opentelemetry-sdk/src/opentelemetry/sdk/_metrics/view.py

```diff
@@ -0,0 +1,172 @@
+           stream can be customized by certain attributes of the corresponding
+           view.
+
+        The attributes documented next serve one of the previous two purposes.
+
+        Args:
+            instrument_type: This is an instrument matching attribute: the class
+            the instrument must be to match the view.
+            instrument_name: This is an instrument matching attribute: the name
+            the instrument must have to match the view.
+            meter_name: This is an instrument matching attribute: the name
+            the instrument meter must have to match the view.
+            meter_version : This is an instrument matching attribute: the
+            version the instrument meter must have to match the view.
+            meter_schema URL : This is an instrument matching attribute: the
+            schema URL the instrument meter must have to match the view.
+            name: This is a metric stream customizing attribute: the name of
+            the metric stream. If `None`, the name of the instrument will be
+            used.
+            description: This is a metric stream customizing attribute: the
+            description of the metric stream. If `None`, the description of the
+            instrument will be used.
+            attribute_keys: This is a metric stream customizing attribute: this
+            is a set of attribute keys. If not `None` then only the measurement
+            attributes that are in `attribute_keys` will be used to identify
+            the metric stream.
+            aggregation: This is a metric stream customizing attribute: every
+            metric stream has an aggregation instance, this is the class of
+            aggregation this instance will be. If `None` the default
+            aggregation class of the instrument will be used.
+        """
+
+        if (
+            instrument_type
+            is instrument_name
+            is meter_name
+            is meter_version
+            is meter_schema_url
+            is None
+        ):
+            raise Exception(
+                "Some instrument selection criteria must be provided"
+            )
+
+        self._instrument_type = instrument_type
+        self._instrument_name = instrument_name
+        self._meter_name = meter_name
+        self._meter_version = meter_version
+        self._meter_schema_url = meter_schema_url
+
+        self._name = name
+
+        if self._name is not None:
+            if self._instrument_type is not None:
+                _logger.warning(
+                    "Instrument type matching disabled "
+                    "for view with a defined name"
+                )
+                self._instrument_type = None
+
+            if self._instrument_name is not None:
+                _logger.warning(
+                    "Instrument name matching disabled "
+                    "for view with a defined name"
+                )
+                self._instrument_name = None
```

Comment:
This case would be OK though, a view with `View(instrument_name="foo", name="bar")` would simply rename. The error condiition is "instrument selectors that select more than one instrument (e.g. wild card instrument name)."

Reply:
Sorry, what do you mean with rename? A view doesn't rename, it selects instruments, and it can select more than one instrument named `"foo"` in that example, right?

Reply:
Ah you're right it could select multiple instruments under different Meters. I opened an issue in the spec because I think this was unintentional https://github.com/open-telemetry/opentelemetry-specification/issues/2337

Reply:
Ok, good, can this conversation be solved or do we need a change?